cassandra/pool.py

```diff
@@ -694,12 +694,15 @@ def _open_connection_to_missing_shard(self, shard_id):
                 return
         shard_aware_endpoint = self._get_shard_aware_endpoint()
         log.debug("shard_aware_endpoint=%r", shard_aware_endpoint)
-
         if shard_aware_endpoint:
-            conn = self._session.cluster.connection_factory(shard_aware_endpoint, on_orphaned_stream_released=self.on_orphaned_stream_released,
-                                                            shard_id=shard_id,
-                                                            total_shards=self.host.sharding_info.shards_count)
-            conn.original_endpoint = self.host.endpoint
+            try:
+                conn = self._session.cluster.connection_factory(shard_aware_endpoint, on_orphaned_stream_released=self.on_orphaned_stream_released,
+                                                                shard_id=shard_id,
+                                                                total_shards=self.host.sharding_info.shards_count)
+                conn.original_endpoint = self.host.endpoint
+            except Exception as exc:
+                log.error("Failed to open connection to %s, on shard_id=%i: %s", self.host, shard_id, exc)
+                raise exc
```

Comment:
Wouldn't it be better to just `raise`? that way you will preserve original backtrace iiuc